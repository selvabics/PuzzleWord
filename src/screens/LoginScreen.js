/**
 * This is Login screen
 * If user will be register for the first time
 * From next time user will be logged in 
 * And they can play the game where the left
 */
import React, { useCallback, useState } from 'react';
import { StyleSheet, Text, TextInput, KeyboardAvoidingView, Alert, Platform } from 'react-native';
import { responsiveHeight, responsiveWidth } from 'react-native-responsive-dimensions';
import { useDispatch } from 'react-redux';
import * as Animated from 'react-native-animatable';

import { navigate } from '../router/navigation';
import { Colors } from '../utils/constants';
import CommonStyles from '../utils/styles';
import { loginUserAction } from '../redux/action/loginActions'
import BackgroundWallper from '../components/backgroundWallpaper';
import ActionButton from '../components/actionButton';

const LoginScreen = () => {
    const dispatch = useDispatch();
    const [email, setEmail] = useState(''); // User email id

    /**
     * This method is called when user click on sign in button 
     */
    const signInButtonClick = useCallback(async () => {
        // If user email id is not valid then show alert message
        const emailRegex = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
        if (!emailRegex.test(email)) {
            Alert.alert('Info', 'Please enter a valid email adderss');
            return;
        }
        // Call actions to login the user into the app
        await dispatch(loginUserAction(email));
        navigate('homeScreen');
    }, [email])

    return (
        <BackgroundWallper>
            <Animated.View animation={'bounceIn'}
                delay={300}
                style={[CommonStyles.flex, CommonStyles.center]}
            >
                <KeyboardAvoidingView behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
                    style={styles.containerView}>
                    <Text style={styles.headerText}>Puzzle Word</Text>
                    <TextInput
                        style={styles.textInputStyle}
                        autoCapitalize={'none'}
                        placeholder="Email Address"
                        keyboardType="email-address"
                        onChangeText={email => setEmail(email)}
                        value={email}
                    />
                    <ActionButton
                        title={'SIGN IN'}
                        onPress={signInButtonClick}
                    />
                </KeyboardAvoidingView>
            </Animated.View>
        </BackgroundWallper>
    );
};

const styles = StyleSheet.create({
    containerView: {
        justifyContent: 'center',
        alignItems: 'center',
    },
    headerText: {
        ...CommonStyles.h1,
        margin: responsiveWidth(3),
        color: 'white',
    },
    textInputStyle: {
        ...CommonStyles.h5,
        marginTop: responsiveHeight(3),
        height: responsiveHeight(6),
        width: responsiveWidth(70),
        borderColor: Colors.secondary,
        borderWidth: 1,
        borderRadius: 5,
        padding: 5,
        textAlign: 'center',
        backgroundColor: Colors.transparentWhite,
    },
    buttonStyle: {
        ...CommonStyles.center,
        height: responsiveHeight(6),
        width: responsiveWidth(70),
        marginTop: responsiveHeight(3),
        padding: responsiveWidth(1),
        borderColor: Colors.primary,
        borderWidth: 1,
        borderRadius: 5,
        backgroundColor: Colors.primary
    }
});

export default LoginScreen;
