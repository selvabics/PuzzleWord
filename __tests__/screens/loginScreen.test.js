import renderer from 'react-test-renderer';
import LoginScreen from '../../src/screens/LoginScreen';

import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store';

jest.useFakeTimers();

describe('Testing login screen', () => {
    it('renders Login Screen', () => {
        const mockStore = configureStore()
        const store = mockStore({})
        const tree = renderer
            .create(<Provider store={store}><LoginScreen /></Provider>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

