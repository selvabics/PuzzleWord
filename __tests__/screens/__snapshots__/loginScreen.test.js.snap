// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing login screen renders Login Screen 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    [
      {
        "flex": 1,
        "paddingBottom": 20,
        "paddingTop": 30,
      },
      {},
    ]
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/images/background.jpg",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    collapsable={false}
    forwardedRef={[Function]}
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0,
        "transform": [
          {
            "scale": 0.3,
          },
        ],
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 61.191865835619396,
            "margin": 22.5,
          }
        }
      >
        Puzzle Word
      </Text>
      <TextInput
        autoCapitalize="none"
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Email Address"
        style={
          {
            "backgroundColor": "rgba(255,255,255,0.4)",
            "borderColor": "rgb(45,70,90)",
            "borderRadius": 5,
            "borderWidth": 1,
            "fontSize": 30.595932917809698,
            "height": 80.03999999999999,
            "marginTop": 40.019999999999996,
            "padding": 5,
            "textAlign": "center",
            "width": 525,
          }
        }
        value="selva@gmail.com"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          collapsable={false}
          forwardedRef={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(136,213,66,0.8)",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 80.03999999999999,
              "justifyContent": "center",
              "margin": 20,
              "opacity": 0,
              "padding": 7.5,
              "transform": [
                {
                  "scale": 0.3,
                },
              ],
              "width": 525,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 30.595932917809698,
                "fontWeight": "bold",
              }
            }
          >
            SIGN IN
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
